SUASSystem/SUASSystem/sda.py

```diff
@@ -2,44 +2,94 @@
-                except:
-                    sda_avoid_coords.append(sda_converter.get_uav_avoid_coordinates())
+        if True:#sda_status.value == "connected":
+            try:
+                current_location = vehicle_state_data[0].get_location()
+                current_location.alt = current_location.get_alt() - SUASSystem.GCSSettings.MSL_ALT
+                current_waypoint_number = vehicle_state_data[0].get_current_waypoint_number()
+
+                if current_waypoint_number != 0:
+                    current_uav_waypoint = waypoints[current_waypoint_number - 1]
+                    current_uav_waypoint_location = SUASSystem.Location(current_uav_waypoint.x, current_uav_waypoint.y, current_uav_waypoint.z * 3.28084)
+
+                    print(current_uav_waypoint)
+                    print("IS WAYPOINT AVOIDABLE: " + str(is_waypoint_avoidable(current_uav_waypoint)))
+                    if is_waypoint_avoidable(current_uav_waypoint) and not is_uav_at_location(current_location, current_uav_waypoint_location):
+                        sda_converter.set_waypoint(current_uav_waypoint_location)
+
+                        #sda_converter.reset_obstacles()
+                        for stationary_obstacle in mission_information_data[0]["stationary_obstacles"]:
+                            sda_converter.add_obstacle(stationary_obstacle)
+                        #for moving_obstacle in mission_information_data["moving_obstacles"]:
+                        #    sda_converter.add_obstacle(get_obstacle_location(moving_obstacle, MSL_ALT), moving_obstacle)
+                        sda_converter.set_uav_position(current_location)
+                        if not UAV_status.value == "GUIDED":
+                            sda_converter.avoid_obstacles()
+                        print('current xy coords')
+                        print(SUASSystem.convert_to_point(starting_coords, current_location))
+                        print('current sda avoid coords')
+                        print(sda_avoid_coords)
+                        print("has completed guided path?")
+                        print(sda_converter.has_uav_completed_guided_path())
+
+                        if not sda_converter.has_uav_completed_guided_path():
+                            UAV_status.value = "GUIDED"
+                            try:
+                                sda_avoid_coords[0] = sda_converter.get_uav_avoid_coordinates()
+                            except:
+                                sda_avoid_coords.append(sda_converter.get_uav_avoid_coordinates())
+
+                        if sda_converter.has_uav_completed_guided_path():
+                            UAV_status.value = "AUTO"
+                            sda_converter.current_path = numpy.array([])
+            except:
+                traceback.print_exc()
 
         sleep(0.5)
+
+def is_waypoint_avoidable(uav_waypoint):
+    non_avoidable_waypoint_command_types = [22, 183, 20]
```

Comment:
Perhaps this list should be placed into the ```GCSSettings``` class?

Reply:
ehhh